torch/cuda/__init__.py

```diff
@@ -225,12 +225,12 @@ def __enter__(self):
         self.prev_idx = torch._C._cuda_getDevice()
         if self.prev_idx != self.idx:
             torch._C._cuda_setDevice(self.idx)
-        _lazy_init()
+        if not torch.jit.is_scripting():
+            _lazy_init()
 
-    def __exit__(self, *args):
+    def __exit__(self, type: Any, value: Any, traceback: Any):
         if self.prev_idx != self.idx:
             torch._C._cuda_setDevice(self.prev_idx)
-        return False
```

Comment:
For JIT compilation, `__exit__` does not support return values. By default if no return value is provided, the exceptions get handled by the Python interpreter which is same as a `return False`. So the behavior remains unchanged for eager mode even by removing return statement.